src/main/kotlin/brs/services/impl/TransactionProcessorServiceImpl.kt

```diff
@@ -108,17 +108,18 @@ class TransactionProcessorServiceImpl(private val dp: DependencyProvider) : Tran
         if (!transaction.verifySignature()) {
             throw BurstException.NotValidException("Transaction signature verification failed")
         }
-        val processedTransactions = processTransactions(listOf(transaction), null)
```

Comment:
As the processTransactions method also add to the unconfirmed list, so it returns always null

Reply:
Good catch.